### PR TITLE
Add some additional datastore tests to improve coverage

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: pkg/datastore/test/counters.go [hunk 1]
```diff
@@ -65,6 +67,41 @@ func RelationshipCounterOverExpiredTest(t *testing.T, tester DatastoreTester) {
...
+
+	// Run multiple registrations of the counter in parallel and ensure only
+	// one succeeds.
+	var numSucceeded atomic.Int32
+	var numFailed atomic.Int32
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			_, err = ds.ReadWriteTx(context.Background(), func(ctx context.Context, tx datastore.ReadWriteTransaction) error {
```
Comment (Verolop) [inline pkg/datastore/test/counters.go:84]: there seems to be a data race due to all goroutines writing to the shared err variable.
Quick workaround: introducing a new variable inside the goroutine:
``` 
localErr := ds.ReadWriteTx(context.Background(), func(ctx context.Context, tx datastore.ReadWriteTransaction) error {

File: pkg/datastore/test/counters.go [hunk 2]
```diff
@@ -65,6 +67,41 @@ func RelationshipCounterOverExpiredTest(t *testing.T, tester DatastoreTester) {
...
+				return tx.RegisterCounter(ctx, "document", &core.RelationshipFilter{
+					ResourceType: testfixtures.DocumentNS.Name,
+				})
+			})
+			if err != nil {
```
Comment (Verolop) [inline pkg/datastore/test/counters.go:89]: updating the reference to the newly introduced variable:
```
if localErr != nil {
    require.Contains(t, localErr.Error(), "counter with name `document` already registered")

**Review by Verolop — CHANGES_REQUESTED** [review]

**Review by tstirrat15 — APPROVED** [review]
LGTM

File: pkg/datastore/test/counters.go (hunk 1 above)
Comment (josephschorr) [reply pkg/datastore/test/counters.go]: Good catch. Fixed

**Comment by josephschorr** [comment]
Updated

**Review by Verolop — APPROVED** [review]

**Pull request merged**